acp/acp.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+/*
+Package acp utilizes the sourcehub acp module to bring the functionality to defradb, this package also helps
+avoid the leakage of direct sourcehub references through out the code base, and eases in swapping
+between local embedded use case and a more global on sourcehub use case.
+*/
+
+package acp
+
+import (
+	"context"
+
+	"github.com/sourcenetwork/immutable"
+
+	"github.com/sourcenetwork/defradb/logging"
+)
+
+var (
+	log = logging.MustNewLogger("acp")
+
+	NoACPModule = immutable.None[ACPModule]()
```

Comment:
todo: Please remove this.  Part of `immutable.Option`'s reason to be is that the default value is `None` and doesn't require this kind of thing.

Reply:
I have mixed feelings about this one. It's just a variable that makes it cleaner to indicate `NoACPModule`.

Reply:
It is an exported/public-internal variable though, not a private convenience thing.

In the code that references this it likely obfuscates what it is, forcing readers to navigate through to see what it is, especially given that it is currently undocumented.

Reply:
I really like it, I can document it if you like. Unless others also have this same preference I will keep it.